### DISCUSSION TITLE
Configurable naturalize_function for NaturalOrderingField

### DISCUSSION THREAD

**Discussion by jiuka** [netbox-community/netbox#14837] [Ideas]
Hi,

We stumbled across a problem where the `naturalize_interface` function leads not to the desired sort order. Updating the sort order would lead to a lot of regressions for others. For (some) ChoiceSet it is possible to [append/owerwrite](https://demo.netbox.dev/static/docs/configuration/data-validation/#field_choices) the available choices. I wounder if it would benefitial to have a similar configuration for `NATURALIZE_FUNCTION` to allow everyone individually to overwrite the function on a per model basis if needed. The configuration could look somting like this:

```
def silly_order(i):
    return f"{i.last_updated.timestamp()}"

NATURALIZE_FUNCTION= {
    'dcim.Interface': silly_order
}
```

Would this be a helfull feature for other too?

**Comment by jeremystretch** [comment]
This is not something we'd implement in core, as it would be infeasible to ensure predictable behavior with configurable logic. However, you're free to fork the code base and implement your own logic locally.

**Reply by jiuka** [reply]
Hi Jeremy,

Can you, or anyone else for that mather, think of any other way it would be possible to provide more flexebility in the sorting order? We try to make use of the render-config feature but strugle with the sort order of the interfaces. (Te1/0/1 is always behind Gi1/0/4). While possible to add a custom sort order field as custom field and maintaining it with events and scripts, i would prefer if there would be a nicer and cleaner solution. There has already been a discussion in issue #9368 with the [conclusion](https://github.com/netbox-community/netbox/issues/9368#issuecomment-1135977690) that the current sort scheme can not be changed. However it would be appreciated to have a clean way of addopting a different sorting if the one settled on does not work. Especialy now with the render-config feature. Befor this feature it did not bother us that the gui showd the interfaces out of order in some cases. Keeping track of a own fork is currently not something i like to do. 

Regards Marius

**Reply by PieterL75** [reply]
I've been waiting for this for several years :-)
I don't like the enforced sort order, as it is now. This is not working in a lot of cases..

In my opinion, the order of th interfaces should be set in a template, with the ability to manually move the order of the interfaces yourself.
All subsequent created devices will have the correct order.
Even at device level, it would be awesome to be able to move interface order.
Our netapp, juniper, arista and other switches would look soooo much nicer.

Unfortunately, I'm a scripter and not a coder. My skills are not good enough to even start implementing this. I have an idea how it could look like, but that's it..
